x/stakeibc/keeper/icacallbacks_reinvest.go

```diff
@@ -115,15 +115,18 @@ func ReinvestCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack
 	// Submit an ICQ for the rewards balance in the fee account
 	k.Logger(ctx).Info(utils.LogICACallbackWithHostZone(chainId, ICACallbackID_Reinvest, "Submitting ICQ for fee account balance"))
 
+	timeout := time.Unix(0, int64(strideEpochTracker.NextEpochStartTime))
+	timeoutDuration := timeout.Sub(ctx.BlockTime())
+
 	query := icqtypes.Query{
 		ChainId:         chainId,
 		ConnectionId:    hostZone.ConnectionId,
 		QueryType:       icqtypes.BANK_STORE_QUERY_WITH_PROOF,
 		RequestData:     queryData,
 		CallbackModule:  types.ModuleName,
 		CallbackId:      ICQCallbackID_FeeBalance,
-		TimeoutDuration: time.Hour,
-		TimeoutPolicy:   icqtypes.TimeoutPolicy_RETRY_QUERY_REQUEST,
+		TimeoutDuration: timeoutDuration,
```

Comment:
I believe this timeout should be a relative amount of time into the future from now (e.g. time.Hour). But here it's being set to the absolute time one block before the start of the next epoch. Is that right?

Reply:
it's being set to the relative time before the epoch. The calculation is:
relativeTimeout = epochEndTime - currentTime

Reply:
Oh duh, I missed `.sub(ctx.blocktime())`

Reply:
All good here!

Reply:
There's actually a TImeout field on the query object that represents the absolute time and is set `SubmitICQRequest`

One thing we could do it allow the specification of _either_ Timeout or TimeoutDuration

Reply:
might make this code a little simpler, but the submission code a little messier

Reply:
leaving this as is